independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/ClassLoadingResource.java

```diff
@@ -5,13 +5,35 @@
 
 public interface ClassLoadingResource {
 
+    /**
+     * A buffer that contains the data of a resource
+     */
+    interface Buffer {
```

Comment:
I could avoid as a whole this interfaces (which will reduce the amount of class checks performed by the generics APIs here and there)